plugin/src/main/kotlin/org/gradle/github/dependency/extractor/internal/json/GitHubDependency.kt

```diff
@@ -1,9 +1,9 @@
 package org.gradle.github.dependency.extractor.internal.json
 
 data class GitHubDependency(
-    val package_url: String?,
-    val relationship: Relationship,
-    val dependencies: List<String>
+    val package_url: String,
+    var relationship: Relationship,
+    val dependencies: MutableList<String> = mutableListOf()
 ) {
```

Comment:
Grumble, grumble.... immutable objects.... grumble, grumble

Kotlin data classes all have a `copy` method on them that is automatically generated that lets you modify only targeted fields.

At the end of the day, not a huge deal, but I prefer to try to keep as many objects immutable as possible since it makes their state through an application easier to reason about.

Reply:
I concur. It would be nice to see the code refactored to work with immutable objects.

Reply:
Yep, this will involve collecting the data into an intermediate builder type and only construct an immutable `GitHubDependency` once all state is known. I was being lazy avoiding this complexity.

Reply:
Done: a1afe9d6e47ee12a4f9951b74a006fbff69eee09